src/ophys_etl/pipelines/segment_binarize_pipeline.py

```diff
@@ -1,54 +1,101 @@
+import json
+import pathlib
+import tempfile
+
+import marshmallow
 import argschema
-from ophys_etl.transforms.suite2p_wrapper import Suite2PWrapper
-from ophys_etl.transforms.convert_rois import BinarizerAndROICreator
-from ophys_etl.schemas.fields import H5InputFile
 
-import tempfile
-import pathlib
-import json
+
+from ophys_etl.transforms.convert_rois import (
+    BinarizeAndCreateROIsInputSchema, BinarizerAndROICreator)
+from ophys_etl.transforms.suite2p_wrapper import (Suite2PWrapper,
+                                                  Suite2PWrapperSchema)
 
 
 class SegmentBinarizeSchema(argschema.ArgSchema):
-    motion_corrected_video = H5InputFile(
-        required=True,
-        description="hdf5 motion-correced video")
-    motion_correction_values = argschema.fields.InputFile(
-        required=True,
-        description="motion correction values in a csv")
+    suite2p_args = argschema.fields.Nested(Suite2PWrapperSchema,
+                                           required=True)
+    convert_args = argschema.fields.Nested(BinarizeAndCreateROIsInputSchema,
+                                           required=False)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.tmpdir = None
+        self.tmp_tempfile = None
+
+    @marshmallow.pre_load
+    def setup_default_suite2p_args(self, data: dict, **kwargs) -> dict:
+
+        if "output_dir" not in data["suite2p_args"]:
+            self.tmpdir = tempfile.TemporaryDirectory()
+            data["suite2p_args"]["output_dir"] = self.tmpdir.name
+
+        if "output_json" not in data["suite2p_args"]:
+            Suite2p_output = (pathlib.Path(data["suite2p_args"]["output_dir"])
+                              / "Suite2P_output.json")
+            data["suite2p_args"]["output_json"] = str(Suite2p_output)
+
+        if "bin_size" not in data["suite2p_args"]:
+            data["suite2p_args"]["bin_size"] = 115
+
+        data["suite2p_args"]["log_level"] = data["log_level"]
+
+        if "convert_args" in data:
```

Comment:
`convert_args['log_level']` inheritance in this case:
```
{
  'log_level': "INFO",
  'convert_args': {
    <does not specify log_level>
  }
}
```

Reply:
Good catch!